### ISSUE TITLE
Notebooks deleted after file system sync

### ISSUE THREAD

**Issue by niels-anders** [laurent22/joplin#5822]
I really enjoy using Joplin as my default notebook application, so thanks for the good work. 

I have set up a local file system sync to a folder that is kept in sync across multiple devices (NAS, mobile phone) using syncthing. I only write notes using joplin on my laptop, the other devices are merely for backup purposes

However, I noticed all my notebooks have been deleted except for two, and one ended up in a red `conflict` notebook. I don't think I can blame syncthing for it since it is a one way sync. But I can't rule it out completely.

## Environment

Joplin version: 2.3.5
Platform: Linux
OS specifics: Kubuntu 20.04

## Steps to reproduce

1.  setup local file system sync
2.  setup syncthing to sync to different device
3.  create notebooks that are automatically synced to folder

## Describe what you expected to happen

I would expect the notebooks to kept in sync, and not being deleted.

## Logfile

I attached my log.txt. 

Sync issues start with/at (I didn't realize this at the time):

``` 
2021-11-18 09:41:50: Synchronizer: "Error: Fail-safe: Sync was interrupted because 100% of the data (68 items) is about to be deleted.
```

and many days later at `2021-12-03 09:42:38` it starts to delete everything anyway (except the most recent notebooks):

[log.txt](https://github.com/laurent22/joplin/files/7648661/log.txt)

**Comment by niels-anders** [comment]
Now that I think of it, It might be related to [this issue about changing the sync folder location](https://github.com/laurent22/joplin/issues/5553) because around that november 18 I upgraded my NAS firmware and moved to a dockerized syncthing installation that could have been messing up the joplin files in that folder (except that everything else is just fine). In any case, the fail-safe was active for two weeks, and then decided to continue anyway.

**Comment by niels-anders** [comment]
The issue is still there. I switched off the file system sync to prevent it happening again. My notes are lost though which is a pity. Luckily there wasn't anything too important in there, but file sync doesn't seem too safe to me. Even when 3rd party tools like syncthing empties the backup folder, it should create new backup files instead of deleting all my notes.

**Comment by niels-anders** [comment]
Having said that, I am aware its not really a backup folder but rather a sync folder, and that would involve deleting too so in that sense it is expected behavior. At first the failsafe worked, but after a while it didn't anymore. Maybe because I made new notes so that some failsafe threshold was not passed anymore. It would have been helpful if this failsafe warning would be more prominent. I didn't notice it before my notes were already gone and I looked at the logs
